llvm/lib/MC/MCParser/AsmParser.cpp

```diff
@@ -1769,11 +1769,9 @@ bool AsmParser::parseStatement(ParseStatementInfo &Info,
     // Treat '}' as a valid identifier in this context.
     Lex();
     IDVal = "}";
-  } else if (Lexer.is(AsmToken::Star) &&
-             getTargetParser().starIsStartOfStatement()) {
-    // Accept '*' as a valid start of statement.
+  } else if (getTargetParser().tokenIsStartOfStatement(ID.getKind())) {
```

Comment:
LCurly/RCurly should be pushed into the new method as well, they are invalid at the start of a statement on most targets. In-tree exceptions are Hexagon (curlies are used for bundles) and X86 (used for vex/evex prefix).